src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/vedtak/vedtaksperiode/Opphørsperiode.kt

```diff
@@ -118,9 +121,18 @@ private fun finnOpphørsperiodeEtterSisteUtbetalingsperiode(
 
     val erFramtidigOpphørPgaBarnehageplass =
         vilkårsvurdering.personResultater.any {
-            it.vilkårResultater.any {
-                it.vilkårType == Vilkår.BARNEHAGEPLASS && it.søkerHarMeldtFraOmBarnehageplass == true &&
-                    it.periodeTom?.tilForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør() == sisteUtbetalingsperiodeTom
+            val vilkårResultaterPåPerson = it.vilkårResultater
+            val regelsett = vilkårResultaterPåPerson.hentRegelsettBruktIVilkår() ?: return@any false
+
+            vilkårResultaterPåPerson.any {
+                val periodeTom = it.periodeTom
+                val periodeTomForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør =
+                    when (regelsett) {
+                        VilkårRegelsett.LOV_AUGUST_2021 -> periodeTom?.tilForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør()
+                        VilkårRegelsett.LOV_AUGUST_2024 -> periodeTom?.tilForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør2024()
+                    }
+
+                it.vilkårType == Vilkår.BARNEHAGEPLASS && it.søkerHarMeldtFraOmBarnehageplass == true && periodeTomForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør == sisteUtbetalingsperiodeTom
             }
```

Comment:
Dersom vi kun skal se på barnehageplass-vilkåret kunne man vurdert å filtrere på det til å begynne med. Synes også det er lettere å lese om man bytter ut `it` med et mer beskrivende navn.

```suggestion
        vilkårsvurdering.personResultater.any { personResultat ->
            val barnehageplassVilkårResultaterPåPerson = personResultat.vilkårResultater.filter { vilkårResultat -> vilkårResultat.vilkårType == Vilkår.BARNEHAGEPLASS }
            val regelsett = barnehageplassVilkårResultaterPåPerson.hentRegelsettBruktIVilkår() ?: return@any false

            barnehageplassVilkårResultaterPåPerson.any { barnehageplassVilkårResultat ->
                val periodeTom = barnehageplassVilkårResultat.periodeTom
                val periodeTomForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør =
                    when (regelsett) {
                        VilkårRegelsett.LOV_AUGUST_2021 -> periodeTom?.tilForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør()
                        VilkårRegelsett.LOV_AUGUST_2024 -> periodeTom?.tilForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør2024()
                    }

                barnehageplassVilkårResultat.søkerHarMeldtFraOmBarnehageplass == true && periodeTomForskjøvetTomMånedForSisteUtbetalingsperiodePgaFremtidigOpphør == sisteUtbetalingsperiodeTom
            }
        }
```

Reply:
Godt poeng!

Endrer på det 🙌